lib/Interpreter/CppInterOp.cpp

```diff
@@ -941,6 +949,82 @@ namespace Cpp {
     return true;
   }
 
+  void GetClassTemplatedMethods(const std::string& name, TCppScope_t parent,
+                                std::vector<TCppFunction_t>& funcs) {
+
+    auto* D = (Decl*)parent;
+
+    if (!parent || name.empty())
+      return;
+
+    D = GetUnderlyingScope(D);
+
+    llvm::StringRef Name(name);
+    auto& S = getSema();
+    DeclarationName DName = &getASTContext().Idents.get(name);
+    clang::LookupResult R(S, DName, SourceLocation(), Sema::LookupOrdinaryName,
+                          Sema::ForVisibleRedeclaration);
+
+    Cpp_utils::Lookup::Named(&S, R, Decl::castToDeclContext(D));
+
+    if (R.empty())
+      return;
+
+    R.resolveKind();
+
+    for (auto* Found : R)
+      if (llvm::isa<FunctionTemplateDecl>(Found))
+        funcs.push_back(Found);
+  }
+
+  TCppFunction_t
+  BestTemplateFunctionMatch(const std::vector<TCppFunction_t>& candidates,
+                            const std::vector<TemplateArgInfo>& explicit_types,
+                            const std::vector<TemplateArgInfo>& arg_types) {
+
+    for (const auto& candidate : candidates) {
+      auto* TFD = (FunctionTemplateDecl*)candidate;
+      clang::TemplateParameterList* tpl = TFD->getTemplateParameters();
+
+      // template parameter size does not match
+      if (tpl->size() < explicit_types.size())
+        continue;
+
+      else {
```

Comment:
No need of this else here, we `continue`d.

Reply:
done